ludwig/utils/backward_compatibility.py

```diff
@@ -811,8 +811,15 @@ def upgrade_defaults_config_for_gbm(config: ModelConfigDict) -> ModelConfigDict:
     defaults = copy.deepcopy(defaults_ref)
     gbm_feature_types = GBMDefaultsConfig.Schema().fields.keys()
     for feature_type in defaults_ref:
+        # GBM only supports binary, number, category and text features
         if feature_type not in gbm_feature_types:
             del defaults[feature_type]
+            continue
+
+        # Remove encoder, decoder and loss from defaults since they only apply to ECD
+        defaults[feature_type].pop("encoder", None)
```

Comment:
Encoders do apply to GBMs

Reply:
@ksbrar is this something that need to be fixed in the defaults?

Reply:
That's true yeah, ok let me take a look to make sure we allow encoder as key

Reply:
@tgaddair seems like it, will make it a part of this PR

Reply:
@tgaddair I was under the impression they don't, but if not then yes they should be added back.

Reply:
@tgaddair to be clear, we only support one hot and tf-idf for text feature encoders for GBMs right?

Reply:
We support onehot and passthrough for category, ATM. TF-IDF is the only one supported for text for now, but we're going to add more soon.

Note that none of this is in v0.7, we I might now make this a cherry-pick into v0.7.